api/query/search.go

```diff
@@ -131,14 +131,26 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		logger.Infof("Forwarding structured search request to cache: %s", string(data))
 
 		client := sh.api.GetHTTPClient()
-		resp, err := client.Post(url, "application/json", bytes.NewBuffer(data))
+		req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
 		if err != nil {
+			logger.Errorf("Failed to create request to POST %s: %v", url, err)
+			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
+			return
+		}
+		req.Header.Add("Content-Type", "application/json")
+
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Errorf("Error connecting to search API cache: %v", err)
+			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
+			return
+		} else if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
1. No need to have the `else` because of early return.
2. More importantly (though not introduced by this PR), `resp.Body` isn't closed.

```suggestion
		}
		defer resp.Body.Close()
		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```